test/hotspot/gtest/gc/shared/test_ptrQueueBufferAllocator.cpp

```diff
@@ -180,12 +180,18 @@ class BufferNode::TestSupport::ProcessorThread : public JavaTestThread {
   {}
 
   virtual void main_run() {
+    bool shutdown_requested = false;
     while (true) {
       BufferNode* node = _cbl->pop();
       if (node != NULL) {
         _allocator->release(node);
-      } else if (!Atomic::load_acquire(_continue_running)) {
+      } else if (shutdown_requested) {
         return;
+      } else if (!Atomic::load_acquire(_continue_running)) {
+        // To avoid a race that could leave buffers in the list after this
```

Comment:
s/avoid/handle/ (or something similar).

The change does not eliminate (or avoids) the race, but handles the (not logic-breaking) results of the race.

Reply:
```suggestion
        // To handle a benign race that could leave buffers in the list after this
```

Reply:
The race involved is not a benign data race, it's a logic race that's not at all benign.  I could describe the race in more detail.

Reply:
I meant "benign" wrt to what is actually tested (functionality of the ptrqueuebuffer), but you are right about that my suggestion is the wrong wording. Just leave it as is then.

Thomas